example/lib/main.dart

```diff
@@ -1,27 +1,94 @@
 import 'package:flutter/material.dart';
-
-import 'src/ui/device_list.dart';
-import 'src/utils.dart';
+import 'package:flutter_reactive_ble/flutter_reactive_ble.dart';
+import 'package:flutter_reactive_ble_example/src/ble/ble_device_connector.dart';
+import 'package:flutter_reactive_ble_example/src/ble/ble_scanner.dart';
+import 'package:flutter_reactive_ble_example/src/ble/ble_status_monitor.dart';
+import 'package:flutter_reactive_ble_example/src/ui/device_list.dart';
+import 'package:provider/provider.dart';
 
 const _themeColor = Colors.lightGreen;
 
 void main() {
-  FlutterError.onError = (error) {
-    log("An error occurred: ${error.exception}");
-    log(error.stack.toString());
-  };
+  WidgetsFlutterBinding.ensureInitialized();
 
-  runApp(const FlutterReactiveBleExampleApp());
+  final _ble = FlutterReactiveBle();
+  final _scanner = BleScanner(_ble);
+  final _monitor = BleStatusMonitor(_ble);
+  final _connector = BleDeviceConnector(_ble);
+  runApp(
+    MultiProvider(
```

Comment:
Multiprovider needs to be the top node since if we put it within material app you need to add the providers within each route. For now let's not do this